### ISSUE TITLE
OneOf transformation?

### ISSUE THREAD

**Issue by chris-tkinter** [kornia/kornia#1159]
## ❓ Questions and Help
Hi, I didn't see it in the documentation, but I am wondering is there a `OneOf` transformation in the kornia?

For example, in AugLy has
```python

AUGMENTATIONS = [
    imaugs.Blur(),
    imaugs.ColorJitter(**COLOR_JITTER_PARAMS),
    imaugs.OneOf(
        [imaugs.OverlayOntoScreenshot(), imaugs.OverlayEmoji(), imaugs.OverlayText()]
    ),
]
```
which means it would apply only one of the `OverlayOntoScreenshot`, `OverlayEmoji` or `OverlayText`. I think albumentations also has a similar one.

It's relatively easy to implement, but I am just want to check if there actually is an `OneOf` transformation in kornia.

Thanks!

**Comment by shijianjian** [comment]
Hi @chris-tkinter, you may use "random_apply" parameter in kornia containers to reach similar results. https://kornia.readthedocs.io/en/latest/augmentation.container.html

**Comment by edgarriba** [comment]
@shijianjian are you sure it’s the same ? With random_apply if I understand well eventually you could have applied the three ops while with OneOf it just samples one of the provide ops

**Comment by shijianjian** [comment]
> @shijianjian are you sure it’s the same ? With random_apply if I understand well eventually you could have applied the three ops while with OneOf it just samples one of the provide ops

You might pass integer values or a tuple in as well, it is not restricted to Boolean values.

**Comment by edgarriba** [comment]
@shijianjian can we have an example/tutorial of the use of that dual functionality ?

**Comment by chris-tkinter** [comment]
Yeah @shijianjian I think it would be nice to have an example. Thanks!

**Comment by shijianjian** [comment]
@chris-tkinter Totally forgot this. Will add it today.

**Comment by vedal** [comment]
From the documentation in https://kornia.readthedocs.io/en/latest/augmentation.container.html, this issue seems ready to be closed

**Comment by edgarriba** [comment]
@vedal you are right, but i have the opinion that having a separate `OneOf` (or `Select`, `Choose` [better namings]) is more explicit and easy to follow ? /cc @shijianjian

**Comment by edgarriba** [comment]
If anyone interested in this please, re-open. Thanks !
